### ISSUE TITLE
Lifetime may not live long enough

### ISSUE THREAD

**Issue by hellow554** [cobalt-org/liquid-rust#342]
While building [cargo generate](https://github.com/ashleygwilliams/cargo-generate) the following error plopped up

```text
warning: lifetime may not live long enough
  --> /home/op/.cargo/registry/src/github.com-1ecc6299db9ec823/liquid-value-0.18.1/src/scalar.rs:60:54
   |
29 | impl<'s> ScalarCow<'s> {
   |      -- lifetime `'s` defined here
...
60 |             ScalarCowEnum::Str(ref x) => Scalar::new(x.as_ref()),
   |                                                      ^^^^^^^^^^ argument requires that `'s` must outlive `'static`
   |
   = warning: this error has been downgraded to a warning for backwards compatibility with previous releases
   = warning: this represents potential undefined behavior in your code and this warning will become a hard error in the future
   = note: for more information, try `rustc --explain E0729`

warning: lifetime may not live long enough
  --> /home/op/.cargo/registry/src/github.com-1ecc6299db9ec823/liquid-value-0.18.1/src/scalar.rs:60:54
   |
54 |     pub fn as_ref<'r: 's>(&'r self) -> ScalarCow<'r> {
   |                   -- lifetime `'r` defined here
...
60 |             ScalarCowEnum::Str(ref x) => Scalar::new(x.as_ref()),
   |                                                      ^^^^^^^^^^ argument requires that `'r` must outlive `'static`
   |
   = warning: this error has been downgraded to a warning for backwards compatibility with previous releases
   = warning: this represents potential undefined behavior in your code and this warning will become a hard error in the future
```

liquid-rust version: apparently 0.18.1
rust version: rustc 1.37.0-nightly (0beb2ba16 2019-07-02)

**Comment by epage** [comment]
Yikes.  Looks like this is one of the soundness bugs that NLL exposed now that they are applying it to 2015 edition.  I'll prioritize looking into this.

**Comment by epage** [comment]
Building now with 1.36 and looks like I'm not the only one
```
warning[E0713]: borrow may still be in use when destructor runs
   --> /home/epage/.cargo/registry/src/github.com-1ecc6299db9ec823/url-1.7.0/src/form_urlencoded.rs:261:40
    |
259 | impl<'a> Target for ::UrlQuery<'a> {
    |      -- lifetime `'a` defined here
260 |     fn as_mut_string(&mut self) -> &mut String { &mut self.url.serialization }
261 |     fn finish(self) -> &'a mut ::Url { self.url }
    |                                        ^^^^^^^^ - here, drop of `self` needs exclusive access to `*self.url`, because the type `UrlQuery<'_>` implements the `Drop` trait
    |                                        |
    |                                        returning this value requires that `*self.url` is borrowed for `'a`
    |
    = warning: this error has been downgraded to a warning for backwards compatibility with previous releases
    = warning: this represents potential undefined behavior in your code and this warning will become a hard error in the future
```

**Comment by epage** [comment]
Thankfully it looks like there is no problem in the code, meaning people are safe to run with the current version and the fix is trivial.  Via the soundness hole, we elevated a lifetime to `'static` but we then coerced it to the right lifetime.  This probably happened when doing a compiler-guided refactoring :)

**Comment by epage** [comment]
liquid v0.19 is now available (there were already some breaking changes queued up).

I'll try to get cargo-generate updated.
